src/main/java/uk/gov/hmcts/reform/pip/data/management/service/helpers/listmanipulation/CrownWarnedPddaListHelper.java

```diff
@@ -0,0 +1,127 @@
+package uk.gov.hmcts.reform.pip.data.management.service.helpers.listmanipulation;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import uk.gov.hmcts.reform.pip.data.management.models.templatemodels.crownpddalist.CrownWarnedPddaList;
+
+import java.time.DayOfWeek;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.temporal.TemporalAdjusters;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Optional;
+
+public final class CrownWarnedPddaListHelper {
+    public static final DateTimeFormatter DATE_FORMATTER = DateTimeFormatter.ofPattern("dd/MM/yyyy");
+
+    private CrownWarnedPddaListHelper() {
+    }
+
+    @SuppressWarnings("PMD.UseConcurrentHashMap")
+    public static Map<String, List<CrownWarnedPddaList>> processPayload(JsonNode warnedPddaListData) {
+        Map<String, List<CrownWarnedPddaList>> groupedData = new LinkedHashMap<>();
+
+        JsonNode warnedList = warnedPddaListData.get("WarnedList");
+        JsonNode courtLists = warnedList.get("CourtLists");
+
+        Optional.ofNullable(courtLists)
+            .filter(JsonNode::isArray)
+            .ifPresent(lists -> lists.forEach(courtList -> {
+                // Process WithFixedDate
+                Optional.ofNullable(courtList.get("WithFixedDate"))
+                    .filter(JsonNode::isArray)
+                    .ifPresent(withFixedDate -> withFixedDate.forEach(withFixDate ->
+                        formatFixture(withFixDate, groupedData, false)));
+
+                // Process WithoutFixedDate
+                Optional.ofNullable(courtList.get("WithoutFixedDate"))
+                    .filter(JsonNode::isArray)
+                    .ifPresent(withoutFixedDate -> withoutFixedDate.forEach(withoutFixDate ->
+                        formatFixture(withoutFixDate, groupedData, true)));
+            }));
+
+        // Sort cases by fixed date within each group
+        groupedData.forEach((key, cases) -> {
+            cases.sort(Comparator.comparing(CrownWarnedPddaList::getFixedDateAsLocalDate,
+                                            Comparator.nullsLast(Comparator.naturalOrder())));
+        });
+
+        return groupedData;
+    }
+
+    private static void formatFixture(JsonNode fixtureDate, Map<String,
+        List<CrownWarnedPddaList>> groupedData, boolean isWithoutFixedDate) {
+        JsonNode fixture = fixtureDate.get("Fixture");
+        String fixedDate = fixture.has("FixedDate") ? fixture.get("FixedDate").asText() : null;
```

Comment:
Using 'findAndReturnNodeText()' might work here to save needing to do the if statement.

Same for the 'HearingDescription,  'ListNote' and 'CaseNumberCaTH' further down